src/store/actions/authentication.js

```diff
@@ -0,0 +1,54 @@
+import swal from 'sweetalert';
+
+function showAlert(text, icon) {
+  swal({
+    title: 'Authors Haven',
+    text,
+    icon,
+    button: 'Okay!',
+  });
+}
+export const requestReset = email => {
+  const mydata = {
+    email,
+  };
+  fetch('http://127.0.0.1:8000/api/password_reset/', {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(mydata),
+  })
+    .then(res => res.json())
+    .then(data => {
+      const text = data.message;
+      if (data.status === 1) {
+        const icon = 'success';
+        showAlert(text, icon);
+      } else {
+        const icon = 'warning';
+        showAlert(text, icon);
+      }
+    });
+};
+
+export const resetPassword = body => {
+  fetch('http://127.0.0.1:8000/api/set_password/complete/', {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(body),
+  })
+    .then(res => res.json())
```

Comment:
The first then of the `fetch` API only guarantees that the browser was able to make a request irrespective of what the result or status of the request is, it will resolve for both the success and error status codes . if you want to check if the request was successful, you have to check for the `ok` property and only proceed if it's true, if false, it would mean something went wrong , say there was a validation issue or you are hitting a  `non` existing `url` at which point you have to bail out, something like this.

```js
. then(res => res.json().then(data => (res.ok ? Promise.resolve(data) : Promise.reject(data))));
```
Clearly it's painful to do these sorts of checks every time you want to make a request, luckily, we have a wrapper that abstracts this sort of duplication in `src/utils/Http` and is injected in all our actions by `redux-thunk`

Reply:
Roland, calling the wrapped class for fetching which would even save me doing `fetch` time and again gave me `content-type` issues that is why I reseorted to the individuals. However, if you could help me find out why the error exists even when the `content-type` was specified I would appreciate